spec/admin/validators/file_validator_spec.rb

```diff
@@ -2,17 +2,43 @@ module Admin
   module Validators
     describe FileValidator do
       context 'Invalidates empty file' do
-        subject { FileValidator.new([]) }
+        subject { FileValidator.new([], []) }
 
         it 'Deems invalid' do
           expect(subject.valid?).to eq(false)
         end
 
         it 'Error message' do
           subject.valid?
-          expect(subject.errors).to eq(['The file contains no data'])
+          expect(subject.errors).to eq(['The file contains no mediator data'])
+        end
+      end
+
+      context 'Blacklist' do
+        let(:mediators) { [ { foo: 42, bar: 43 } ] }
+
+        context 'Found in data' do
+          subject { FileValidator.new(mediators, %w{ foo }) }
+
+          it 'Deems valid' do
+            expect(subject.valid?).to eq(true)
+          end
+        end
+
+        context 'Not found in data' do
```

Comment:
Might be worth having a test to cover when there's more than 1 blacklisted field that isn't present in the mediator keys.